### PR TITLE
Changes to travis config

### PR DESCRIPTION
**Pull request by chebizarro**
This PR hopefully addresses build errors with the removal of old Python versions. The changes here have passed the validator at https://config.travis-ci.com/explore with deprecated tags removed and the distribution updated to bionic (18.04).

### CODE REVIEW

**Comment by peterjc** [comment]
Looks like that has got TravisCI to do something following the silent failure in #4.

**Comment by peterjc** [comment]
Hmm, ``ModuleNotFoundError: No module named 'pip.pep425tags'`` - I'd suggest looking over the https://github.com/matthew-brett/multibuild issues (including closed issues). Hopefully they've fixed this already...

**Comment by peterjc** [comment]
I usually look here first as the main showcase for multibuild,
https://github.com/MacPython/astropy-wheels/blob/master/.travis.yml

They have:

```
language: python
# Default Python version is usually 2.7
python: 3.6
dist: trusty
```

**Comment by matthew-brett** [comment]
Actually - I'd recommend updating the lastest `devel` branch of multibuild - it's the most up-to-date.

**Comment by peterjc** [comment]
I think the state of the biopython submodule is immaterial - the scripts pull the specified commit from GitHub anyway.

Whenever we update multibuild, there is almost always some tweaks needed to the AppVeyor or TravisCI files. In case it matters, did you update to their latest master or devel branch?

(Update: To clarify, this was not meant negatively - just usually I've only done the updated when something breaks - and then there is a backlog of changes to apply)

**Comment by chebizarro** [comment]
I did, which fixed the pip error but now it seems to be getting stuck on looking for the deprecated Bio.KDTree package: https://travis-ci.org/github/biopython/biopython-wheels/jobs/690697050 -
`FileNotFoundError: [Errno 2] No such file or directory: '/venv/lib/python3.7/site-packages/Bio/KDTree` which is why I tried resetting the submodule state.

**Comment by peterjc** [comment]
I was going to say maybe we missed something in https://github.com/biopython/biopython/pull/2872 / https://github.com/biopython/biopython/pull/2874 - well spotted.

File: .travis.yml [hunk 1]
```diff
@@ -16,40 +16,42 @@ env:
...
   exclude:
-    # Exclude the default (TravisCI provided) Python 3.5 build
-    - python: 3.5
+    - python: 3.6
   include:
     - os: linux
       env:
         - MB_PYTHON_VERSION=3.6
         - TEST_DEPENDS="numpy==1.10.1"
+      python: 3.6
```
Comment (peterjc) [inline .travis.yml]: I don't think we need these lines - at least, the latest astropy does not have them?

https://github.com/MacPython/astropy-wheels/blob/master/.travis.yml

(But let's get the main stumbling block fixed first)

**Comment by peterjc** [comment]
Why bump to bionic? The rule of thumb with building cross-platform Linux binaries is to compile on the oldest supported system. Also astropy is still using trusty.

The failure on trusty about ``test_PDB_internal_coords `` is from a new test, and could be a real failure our standard CI has missed - see https://github.com/biopython/biopython/pull/2399

File: .travis.yml (hunk 1 above)
Comment (peterjc) [reply .travis.yml]: I think we should remove these -  it is missing on the Python 3.8 entry for 64bit (i686) which was the only Linux target with worked on this build:

https://travis-ci.org/github/biopython/biopython-wheels/jobs/690700231

**Comment by peterjc** [comment]
https://travis-ci.org/github/biopython/biopython-wheels/builds/690700225 using Trusty is very close - one Linux wheel worked (and all the macOS wheels worked).

The broken Linux builds are where you changed the underlying base Python provided by TravisCI, which appears to be a bad idea.

**Comment by chebizarro** [comment]
Yeah, that would be my default too, but the error here - https://travis-ci.org/github/biopython/biopython-wheels/jobs/690700230 - seemed to be distro related, which is why I bumped to Bionic but the Astropy config seems to work for some reason.

**Comment by chebizarro** [comment]
So all of the jobs build now except for the ones running under Python 3.6/Numpy 1.10.1
According to the numpy docs, 1.10.1 only supports Python 3.2-3.5 - https://numpy.org/devdocs/release/1.10.1-notes.html. Did we want to try bumping to a later version that supports Python 3.6?

**Comment by peterjc** [comment]
Good - removing the new "python: 3.x" lines seems to have helped.

Putting the macOS builds first seems sensible given the astropy comments.

As to numpy versions, we've tried to use the oldest supported numpy for each version of Python. Again, aiming for maximum binary compatibility between our wheels and whatever the end user might have installed.

(In case you are wondering, this sort of thing has happened a few times - you'll see traces of it on the commit history, and more on the TravisCI history - it is why I wanted to leave at least a day's margin)

**Comment by peterjc** [comment]
TravisCI is happy - just an AppVeyor timeout on GitHub which ought to be fine on a retry (update: yes, fine).

Ready to merge? I would do it as a squash commit...

**Comment by chebizarro** [comment]
Some of the commits didn't squash easily but it's a lot more compact now, I'm ready to go if it all looks good to you.

**Comment by peterjc** [comment]
It won't rebase automatically, but GitHub will let me squash merge it, so I'll just do that in 10 minutes time - unless you say you'd like to do a manual ``git rebase`` and force push to preserve a bit more history?

**Comment by chebizarro** [comment]
I'm fine if you squash merge it, the end results were pretty straightforward.

**Comment by peterjc** [comment]
Done - that should get the Linux and macOS wheels up onto RackSpace shortly. Back later...

**Comment by peterjc** [comment]
https://mailman.open-bio.org/pipermail/biopython/2020-May/016795.html

**Pull request merged**